lib/src/json_schema/custom_vocabularies.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:json_schema/src/json_schema/typedefs.dart';
+
+/// Register a custom vocabulary with the [JsonSchema] compiler.
+class CustomVocabulary {
+  CustomVocabulary(this._vocab, this._setters);
+
+  Uri _vocab;
+  Map<String, KeywordProcessor> _setters;
+
+  Uri get vocab => _vocab;
+
+  Map<String, KeywordProcessor> get setters => _setters;
+}
+
+/// A class to contain the set of functions for setting and validating keywords in a custom vocabulary.
+class KeywordProcessor {
+  KeywordProcessor(this._setter, this._validator);
+
+  SchemaPropertySetter _setter;
+  CustomValidationResult Function(Object, Object) _validator;
+
+  SchemaPropertySetter get setter => this._setter;
+
+  CustomValidationResult Function(Object, Object) get validator => this._validator;
+}
```

Comment:
This class is used to hold the Setter function for the JsonSchema compiler and the Validation function for the validator. Better name ideas?

Reply:
"Processor" isn't the _worst_ name I can think of. Other words that come to mind are "Plugin" or "Extension". If it's only used for custom keywords, I might throw the word "Custom" in there for clarity.

Reply:
Maybe `CustomKeywordSupport`?

Reply:
`KeywordResolver` / `KeywordImplementation`? I wonder if we could use this class to separate our spec-keywords into files and into vocabularies in a more structured way.

Reply:
I was having a similar thought. It might be possible to remove the Validate class all together.